Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -322,12 +373,25 @@ void OpenALStream::SoundLoop()
 						ERROR_LOG(AUDIO, "Error occurred while buffering float32 data: %08x", err);
 					}
 				}
+				else if (fixed32_capable)
+				{
+					int stereo_int32[OAL_MAX_SAMPLES * STEREO_CHANNELS * OAL_MAX_BUFFERS];
+
+					for (u32 i = 0; i < nSamples * STEREO_CHANNELS; ++i)
+					{
+						// Clamping is not necessary here, samples are always between (-1,1)
+						stereo_int32[i] = (int)((float)sampleBuffer[i] * (1 << 31));
```

Comment:
By my tests sampleBuffer never got close enough to 1.0 for that to be a problem, and I assumed it would work as this is how it was originally written on line 395 (but for 16-bit fixed point). Of course it should have probably been something like (1LL << 31) or (INT64_C(1) << 31).